### DISCUSSION TITLE
Robolectric 4.10 Alpha 1

### DISCUSSION THREAD

**Discussion by hoisie** [robolectric/robolectric#8054] [Announcements]
Robolectric 4.10 adds support for native Android graphics. It is currently disabled by default and can be enabled using `@GraphicsMode(NATIVE)`. When native graphics is enabled, interactions with Android graphics classes use real native Android graphics code and are much higher fidelity.

## What's Changed
* Remove dead code that uses Plan based instrumentation (ClassHandler.Plan) https://github.com/robolectric/robolectric/pull/7610
* Run SDK 33 tests on CI by @utzcoz in https://github.com/robolectric/robolectric/pull/7613
* Bump maximum API for GMD to SDK 33 by @utzcoz in https://github.com/robolectric/robolectric/pull/7614
* Remove use-sdk from androidx_test/sharedTest/AndroidManifest.xml by @utzcoz in https://github.com/robolectric/robolectric/pull/7615
* [Warning] Eliminated AnnotateFormatMethod warnings by @hellosagar in https://github.com/robolectric/robolectric/pull/7353
* Remove 0x prefix for string format log by @utzcoz in https://github.com/robolectric/robolectric/pull/7617
* Bump compile/targetSdk to 33 by @utzcoz in https://github.com/robolectric/robolectric/pull/7616
* Remove unused Exception for ShadowActivityTest by @utzcoz in https://github.com/robolectric/robolectric/pull/7618
* Support Activity setShowWhenLocked() and setTurnScreenOn() by @utzcoz in https://github.com/robolectric/robolectric/pull/7438
* Exclude kotlin-stdlib dependency from utils generated .pom by @utzcoz in https://github.com/robolectric/robolectric/pull/7607
* Move dependencies of shadows/framework to gradle.properties by @utzcoz in https://github.com/robolectric/robolectric/pull/7627
* Bump ASM to 9.3 by @utzcoz in https://github.com/robolectric/robolectric/pull/7626
* Add the ability to deploy specific preinstrumented jars to maven by @hoisie in https://github.com/robolectric/robolectric/pull/7602
* Combine exceptions for ReflectionHelpers#getDeclaredField by @utzcoz in https://github.com/robolectric/robolectric/pull/7634
* Clean up ShadowAccountManager by @utzcoz in https://github.com/robolectric/robolectric/pull/7636
* Migrate Gradle dependencies to dependencies.gradle by @utzcoz in https://github.com/robolectric/robolectric/pull/7635
* Add alternative to Unsafe.defineAnonymousClass by @utzcoz in https://github.com/robolectric/robolectric/pull/7633
* Support all authentication results for BiometricManager by @charlesng in https://github.com/robolectric/robolectric/pull/7427
* Bump bcprov-jdk15on from 1.68 to 1.70 by @dependabot in https://github.com/robolectric/robolectric/pull/7638
* Bump constraintlayout from 2.1.3 to 2.1.4 by @dependabot in https://github.com/robolectric/robolectric/pull/7640
* Reuse bytecode of ClassWriter for ProxyMaker by @utzcoz in https://github.com/robolectric/robolectric/pull/7643
* Migrate instance calling in ReflectionHelpers to lambda by @utzcoz in https://github.com/robolectric/robolectric/pull/7644
* Bump compile-testing from 0.18 to 0.19 by @dependabot in https://github.com/robolectric/robolectric/pull/7645
* Bump apiCompatVersion from 4.7 to 4.8.2 by @dependabot in https://github.com/robolectric/robolectric/pull/7647
* Bump gson from 2.8.6 to 2.9.1 by @dependabot in https://github.com/robolectric/robolectric/pull/7646
* Bump core from 1.5.0 to 1.9.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7649
* Clean up unused code for ShadowDisplayManagerTest by @utzcoz in https://github.com/robolectric/robolectric/pull/7662
* Remove unused shadows ShadowCornerPathEffect and ShadowLinearGradient https://github.com/robolectric/robolectric/pull/7668
* Bump apiCompatVersion from 4.8.2 to 4.9 by @dependabot in https://github.com/robolectric/robolectric/pull/7672
* Add call to $$robo$init for Jacoco-instrumented constructors https://github.com/robolectric/robolectric/pull/7677
* Respect ShadowProvider priority when populating the ShadowPicker map https://github.com/robolectric/robolectric/pull/7686
* Properly parse & populate process information from the manifest. ServiceInfo.processName and ActivityInfo.processName will now correctly reflect the value of the process attribute from their respective manifest entry. https://github.com/robolectric/robolectric/pull/7690
* Bump autoServiceVersion from 1.0-rc6 to 1.0.1 by @dependabot in https://github.com/robolectric/robolectric/pull/7674
* Create an activity context for activities. https://github.com/robolectric/robolectric/pull/6481
* Remove deprecated Utils#intArrayToList by @utzcoz in https://github.com/robolectric/robolectric/pull/7687
* Fix configuration/resource handling in activity contexts https://github.com/robolectric/robolectric/pull/7692
* Add Java source & target compatibility config for preinstrumented by @utzcoz in https://github.com/robolectric/robolectric/pull/7664
* Update the `ResourcesManager` configuration when applying qualifiers https://github.com/robolectric/robolectric/pull/7697
* Replace Config with Implementation https://github.com/robolectric/robolectric/pull/7710
* Switch to ubuntu-20.04 for GitHub Actions by @utzcoz in https://github.com/robolectric/robolectric/pull/7716
* Bump errorproneVersion from 2.9.0 to 2.16 by @dependabot in https://github.com/robolectric/robolectric/pull/7715
* Bump Kotlin to 1.7.10 by @utzcoz in https://github.com/robolectric/robolectric/pull/7497
* Upgrade androidx.test and androidx.fragment versions to latest. by @brettchabot in https://github.com/robolectric/robolectric/pull/7683
* Bump kotlinVersion from 1.5.20 to 1.7.20 by @dependabot in https://github.com/robolectric/robolectric/pull/7694
* Adding proxy settings to fetching artifacts by @sebastienrussell in https://github.com/robolectric/robolectric/pull/7713
* Fix potential ClassCastException in RobolectricModel https://github.com/robolectric/robolectric/pull/7735
* Ignore hprof file for Git by @utzcoz in https://github.com/robolectric/robolectric/pull/7739
* Fix some issues in ShadowWallpaperManager https://github.com/robolectric/robolectric/pull/7740
* Disable animation when running instrumentation tests by @utzcoz in https://github.com/robolectric/robolectric/pull/7741
* Switch to use macos-12 for instrumentation-tests on CI by @utzcoz in https://github.com/robolectric/robolectric/pull/7743
* Make CI Emulator's environment cleaner by @utzcoz in https://github.com/robolectric/robolectric/pull/7742
* Bump auto-value from 1.9 to 1.10 by @dependabot in https://github.com/robolectric/robolectric/pull/7747
* Add shadow implementation of MediaController#getExtras() https://github.com/robolectric/robolectric/pull/7749
* Clean up language that appears to be cursing in Korean by @rexhoffman in https://github.com/robolectric/robolectric/pull/7752
* Create Robolectric shadow for ResponderLocation. https://github.com/robolectric/robolectric/pull/7754
* Make ShadowBitmap public APIs extendable https://github.com/robolectric/robolectric/pull/7756
* Add getPackageUid (String packageName, PackageManager.PackageInfoFlags flags) to ShadowApplicationPackageManager. https://github.com/robolectric/robolectric/pull/7737
* Update setLinkDownstreamBandwidthKbps Shadow method to shadow all APIs and have loose signatures https://github.com/robolectric/robolectric/pull/7758
* Add new APIs to ShadowAlarmManager. https://github.com/robolectric/robolectric/pull/7759
* Update snapshot version to 4.10 in README.md by @utzcoz in https://github.com/robolectric/robolectric/pull/7760
* Fix android-9.0.0_r12 ResourceTypes.h link for ResTable by @utzcoz in https://github.com/robolectric/robolectric/pull/7762
* Update to androidx.test 10.26.2022 RC release by @utzcoz in https://github.com/robolectric/robolectric/pull/7763
* Make ShadowMatrix public APIs extendable https://github.com/robolectric/robolectric/pull/7768
* Treat `Pattern` as the same if the pattern and flags are the same https://github.com/robolectric/robolectric/pull/7776
* Remove unused looseSignatures from ShadowConextHubClient by @utzcoz in https://github.com/robolectric/robolectric/pull/7769
* Update some access modifiers from private to protected to allow for overriding and subclass access https://github.com/robolectric/robolectric/pull/7781
* Enclose The Robolectric.buildActivity() call with in try by @arinmodi in https://github.com/robolectric/robolectric/pull/7782
* fix: Avoid referring to Android S's TelephonyCallback by @styluo in https://github.com/robolectric/robolectric/pull/7786
* Bump AGP to 7.3.1 by @utzcoz in https://github.com/robolectric/robolectric/pull/7789
* Enable Implements validation on Gradle again by @utzcoz in https://github.com/robolectric/robolectric/pull/7788
* Add missed override for LocalActivityInvoker#startActivityForResult by @utzcoz in https://github.com/robolectric/robolectric/pull/7790
* Migrate shadows/playservices to AndroidX by @jselbo in https://github.com/robolectric/robolectric/pull/7784
* Only filter sdk when PREINSTRUMENTED_SDK_VERSIONS is not empty by @utzcoz in https://github.com/robolectric/robolectric/pull/7787
* Use uninterruptibly acquire for ResTable's lock by @utzcoz in https://github.com/robolectric/robolectric/pull/7761
* Change where addCallToRoboInit() inserts call to $$robo$init https://github.com/robolectric/robolectric/pull/7793
* Deletes some ShadowNumberPicker functions that don't do anything, which allows DatePickerSpinner to use real values and the real class. https://github.com/robolectric/robolectric/pull/7797
* Update to androidx.test 11.04.2022 stable release by @utzcoz in https://github.com/robolectric/robolectric/pull/7785
* Add getActiveDataSubscriptionId to ShadowSubscriptionManager. https://github.com/robolectric/robolectric/pull/7821
* Delete unnecessary ShadowOutline https://github.com/robolectric/robolectric/pull/7824
* Bump ASM to 9.4 by @utzcoz in https://github.com/robolectric/robolectric/pull/7823
* Keep dependency the same as robolectric.org for Windows building by @utzcoz in https://github.com/robolectric/robolectric/pull/7825
* Enable openRawResourceFd for ResourcesTest by @utzcoz in https://github.com/robolectric/robolectric/pull/7688
* Add shadow for LocaleManager#getSystemLocales by @utzcoz in https://github.com/robolectric/robolectric/pull/7804
* Make ShadowSensorManager call onFlushCompleted on relevant listeners when SensorManager.flush is called. https://github.com/robolectric/robolectric/pull/7837
* Remove lifecycle-common dependency by @utzcoz in https://github.com/robolectric/robolectric/pull/7838
* Fix package name in AndroidManifest.xml of compat-target28 by @utzcoz in https://github.com/robolectric/robolectric/pull/7843
* Make OBJECT_TYPE in ClassInstrumentor protected and add getter method for classType in MutableClass https://github.com/robolectric/robolectric/pull/7847
* Add test-services for instrumentation-test modules by @utzcoz in https://github.com/robolectric/robolectric/pull/7842
* Use \R as regex's linebreak matcher for ExpectedLogMessagesRuleTest by @utzcoz in https://github.com/robolectric/robolectric/pull/7855
* Remove unused code for ReflectionHelperTest by @utzcoz in https://github.com/robolectric/robolectric/pull/7858
* Rename ShadowNativeAllocationRegistry to ShadowNoopNativeAllocationRegistry https://github.com/robolectric/robolectric/pull/7861
* Add JUnit4 Enclosed runner compatibility tests by @utzcoz in https://github.com/robolectric/robolectric/pull/7873
* Fix format of androidx_test's layout xml files by @utzcoz in https://github.com/robolectric/robolectric/pull/7853
* Disable auto-correct for EspressTest's EditText by @utzcoz in https://github.com/robolectric/robolectric/pull/7854
* Use binary resources for the 'nativeruntime' project by @hoisie in https://github.com/robolectric/robolectric/pull/7874
* Ensure ShadowSpeechRecognizer can work with SDK 28 by @utzcoz in https://github.com/robolectric/robolectric/pull/7859
* Keep to use Ubuntu 18.04 for nativeruntime by @utzcoz in https://github.com/robolectric/robolectric/pull/7881
* Update DefaultNativeRuntimeLoader to work in RNG https://github.com/robolectric/robolectric/pull/7882
* Enable CI when commits are pushed to release branches by @hoisie in https://github.com/robolectric/robolectric/pull/7884
* Add ShadowUIModeManager#setCurrentModeType https://github.com/robolectric/robolectric/pull/7885
* Add Espresso test for button clicking with onLongClickListener by @utzcoz in https://github.com/robolectric/robolectric/pull/7857
* CI: Migrate set-output to GITHUB_OUTPUT by @utzcoz in https://github.com/robolectric/robolectric/pull/7888
* Enable useTestStorageService for androidx_test by @utzcoz in https://github.com/robolectric/robolectric/pull/7889
* Cleanup build by @hoisie in https://github.com/robolectric/robolectric/pull/7883
* Bump libphonenumber from 8.0.0 to 8.13.3 by @dependabot in https://github.com/robolectric/robolectric/pull/7891
* Include :integration_tests:play_services for Android Studio by @utzcoz in https://github.com/robolectric/robolectric/pull/7898
* Bump mockitoVersion from 4.1.0 to 4.11.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7899
* Ignore tests of CI for md only modification by @utzcoz in https://github.com/robolectric/robolectric/pull/7903
* Bump multiple dependencies by @utzcoz in https://github.com/robolectric/robolectric/pull/7902
* Bump compile-testing from 0.19 to 0.21.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7905
* Bump errorproneVersion from 2.16 to 2.17.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7906
* Bump kotlinVersion from 1.7.22 to 1.8.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7912
* Bump okhttp-bom from 4.8.0 to 4.10.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7911
* Bump libphonenumber from 8.13.3 to 8.13.4 by @dependabot in https://github.com/robolectric/robolectric/pull/7913
* Add getCallerConfiguredNetworks and removeNonCallerConfiguredNetworks. https://github.com/robolectric/robolectric/pull/7916
* Bump apiCompatVersion from 4.9 to 4.9.2 by @dependabot in https://github.com/robolectric/robolectric/pull/7926
* Bump errorproneVersion from 2.17.0 to 2.18.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7925
* Bump appcompat from 1.4.1 to 1.6.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7924
* Use loose signatures for ShadowApplicationPackageManager.getProperty https://github.com/robolectric/robolectric/pull/7929
* Add example for custom shadow with Kotlin by @utzcoz in https://github.com/robolectric/robolectric/pull/7901
* Adapt to ViewRootImpl.dispatchResized changes https://github.com/robolectric/robolectric/pull/7932
* Add eglGetCurrentContext() implementation in ShadowEGL14. https://github.com/robolectric/robolectric/pull/7933
* Update some dependencies by @utzcoz in https://github.com/robolectric/robolectric/pull/7935
* Bump espressoVersion from 3.5.0 to 3.5.1 by @dependabot in https://github.com/robolectric/robolectric/pull/7937
* Bump monitor from 1.6.0 to 1.6.1 by @dependabot in https://github.com/robolectric/robolectric/pull/7950
* Support ResTable staged alias type by @jselbo in https://github.com/robolectric/robolectric/pull/7923
* Fix typo for RobolectricShadow by @utzcoz in https://github.com/robolectric/robolectric/pull/7954
* RoboCookieManager supports override cookie has the same key by @utzcoz in https://github.com/robolectric/robolectric/pull/7897
* Bump io.mockk:mockk from 1.9.3 to 1.13.4 by @dependabot in https://github.com/robolectric/robolectric/pull/7955
* Bump gson from 2.9.1 to 2.10.1 by @dependabot in https://github.com/robolectric/robolectric/pull/7936
* Bump bcprov-jdk15on from 1.70 to 1.72 by @brandonweeks in https://github.com/robolectric/robolectric/pull/7961
* Migrate kotlinOptions to compilerOptions by @utzcoz in https://github.com/robolectric/robolectric/pull/7976
* Bump androidx.appcompat:appcompat from 1.6.0 to 1.6.1 by @dependabot in https://github.com/robolectric/robolectric/pull/7966
* Bump androidx.test:runner from 1.5.0 to 1.5.2 by @dependabot in https://github.com/robolectric/robolectric/pull/7965
* Remove --disable-shared from ICU's configuration command for MinGW by @utzcoz in https://github.com/robolectric/robolectric/pull/7979
* Clean up PackageManagerTest code by @utzcoz in https://github.com/robolectric/robolectric/pull/7980
* Bump com.diffplug.spotless:spotless-plugin-gradle from 6.12.0 to 6.15.0 by @dependabot in https://github.com/robolectric/robolectric/pull/7984
* Bump com.google.auto.value:auto-value from 1.9 to 1.10.1 by @dependabot in https://github.com/robolectric/robolectric/pull/7982
* Support PackageManager#getApplicationInfo(String, ApplicationInfoFlags) by @utzcoz in https://github.com/robolectric/robolectric/pull/7981
* Add method to set the picture-in-picture mode flag. https://github.com/robolectric/robolectric/pull/7987
* Bump fragmentVersion from 1.2.0 to 1.5.5 by @dependabot in https://github.com/robolectric/robolectric/pull/7983
* Avoid instrumenting some legacy support packages https://github.com/robolectric/robolectric/pull/7993
* Bump com.ibm.icu:icu4j from 70.1 to 72.1 by @dependabot in https://github.com/robolectric/robolectric/pull/8006
* Bump com.googlecode.libphonenumber:libphonenumber from 8.13.4 to 8.13.7 by @dependabot in https://github.com/robolectric/robolectric/pull/8008
* Bump kotlinVersion from 1.8.0 to 1.8.10 by @dependabot in https://github.com/robolectric/robolectric/pull/8009
* Fix sandboxed class lookup when a test is subclassed. https://github.com/robolectric/robolectric/pull/8003
* Bump com.google.auto.value:auto-value-annotations from 1.10 to 1.10.1 by @dependabot in https://github.com/robolectric/robolectric/pull/8007
* Add shadow implementation of UserManager#getUserHandles(boolean) https://github.com/robolectric/robolectric/pull/8018
* Add integration tests for JaCoCo on-the-fly instrumented. by @ZSmallX in https://github.com/robolectric/robolectric/pull/8017
* Only run jacoco-on-the-fly tests on Robolectric environment by @utzcoz in https://github.com/robolectric/robolectric/pull/8019
* Migrate compileTestKotlin to compileKotlin for kotlin module by @utzcoz in https://github.com/robolectric/robolectric/pull/8020
* Bump com.diffplug.spotless:spotless-plugin-gradle from 6.15.0 to 6.16.0 by @dependabot in https://github.com/robolectric/robolectric/pull/8023
* Bump com.android.tools.build:gradle from 7.4.0 to 7.4.2 by @dependabot in https://github.com/robolectric/robolectric/pull/8026
* Bump org.jetbrains:annotations from 16.0.2 to 24.0.1 by @dependabot in https://github.com/robolectric/robolectric/pull/8022
* Make ShadowCompanionDeviceManagerTest future proof. https://github.com/robolectric/robolectric/pull/8027
* Switch JaCoCo on-the-fly to offline for integration tests. by @ZSmallX in https://github.com/robolectric/robolectric/pull/8021
* Convert test code of maven-dependency-resolver to Kotlin by @utzcoz in https://github.com/robolectric/robolectric/pull/7947
* Add xmlns:app="http://schemas.android.com/apk/res-auto" to use app:actionViewClass with the appcompat library https://github.com/robolectric/robolectric/pull/8043
* Skip ShadowHardwareBufferTest#createInvalidFormatThrows on platforms > T. https://github.com/robolectric/robolectric/pull/8036
* Use prebuilt Robolectric Native Runtime artifacts from AOSP by @hoisie in https://github.com/robolectric/robolectric/pull/8045
* Fix an edge case when PackageManager.getApplicationInfo is called by @hoisie in https://github.com/robolectric/robolectric/pull/8048
* Remove CMake/Clang/GCC requirements from README.md by @utzcoz in https://github.com/robolectric/robolectric/pull/8049
* Add Javadoc for some integration_tests/jacoco-offline sources by @hoisie in https://github.com/robolectric/robolectric/pull/8052
* Fix Kotlin lint issues with MavenDependencyResolverTest by @hoisie in https://github.com/robolectric/robolectric/pull/8053
* Remove setOnLongClickListener call in EspressoActivity by @hoisie in https://github.com/robolectric/robolectric/pull/8051
* Remove unused recursive checkout parameter from GitHub Actions by @utzcoz in https://github.com/robolectric/robolectric/pull/8050

## New Contributors
* @charlesng made their first contribution in https://github.com/robolectric/robolectric/pull/7427
* @dependabot made their first contribution in https://github.com/robolectric/robolectric/pull/7638
* @sebastienrussell made their first contribution in https://github.com/robolectric/robolectric/pull/7713
* @rexhoffman made their first contribution in https://github.com/robolectric/robolectric/pull/7752
* @arinmodi made their first contribution in https://github.com/robolectric/robolectric/pull/7782
* @styluo made their first contribution in https://github.com/robolectric/robolectric/pull/7786
* @brandonweeks made their first contribution in https://github.com/robolectric/robolectric/pull/7961

**Full Changelog**: https://github.com/robolectric/robolectric/compare/robolectric-4.9.2...robolectric-4.10-alpha-1

<hr /><em>This discussion was created from the release <a href='https://github.com/robolectric/robolectric/releases/tag/robolectric-4.10-alpha-1'>Robolectric 4.10 Alpha 1</a>.</em>

**Comment by utzcoz** [comment]
It's the first release after  enabling dependabot. From previous expereince, it works great and help us a lot to reduce the burden of updating Gradle's dependencies.
